go/model/model.go

```diff
@@ -261,19 +261,45 @@ func loadModelFromDB(db *database.DB, table, cwd string) (*Model, error) {
 		}
 		defer os.RemoveAll(cwd)
 	}
-	tarFile, err := DumpDBModel(db, table, cwd)
+	tarFile, err := dumpDBModel(db, table, cwd)
 	if err != nil {
 		return nil, err
 	}
 	if !unzipModel {
-		return ExtractMetaFromTarball(tarFile, cwd)
+		return extractMetaFromTarball(tarFile, cwd)
 	}
 	return loadTar(path.Dir(tarFile), path.Base(tarFile), cwd)
 }
 
-// DumpDBModel dumps a model tarball from database to local
+// DumpDBModelAndExtractMeta dumps db model into a tar file and extracts the metadata
+func DumpDBModelAndExtractMeta(db *database.DB, table, cwd string, getMeta bool) (string, *Model, error) {
```

Comment:
Can we have two functions like :

```go
// load model and meta.
model := NewModelFromDB(db *database.DB, table)
// dump model and meta to a tarball.
model.DumpToTar()
```

This is more straightforward forward maybe.